redundant-bmc/src/services_impl.cpp

```diff
@@ -155,4 +157,37 @@ bool ServicesImpl::getProvisioned() const
     return true;
 }
 
+std::string ServicesImpl::getFWVersion() const
+{
+    std::ifstream versionFile{"/etc/os-release"};
+    std::string line;
+    std::string keyPattern{"VERSION_ID="};
+    std::string version;
+
+    while (std::getline(versionFile, line))
+    {
+        // Handle either quotes or no quotes around the value
+        if (line.substr(0, keyPattern.size()).find(keyPattern) !=
+            std::string::npos)
+        {
+            // If the value isn't surrounded by quotes, then pos will be
+            // npos + 1 = 0, and the 2nd arg to substr() will be npos
+            // which means get the rest of the string.
+            auto value = line.substr(keyPattern.size());
+            std::size_t pos = value.find_first_of('"') + 1;
+            version = value.substr(pos, value.find_last_of('"') - pos);
+            break;
+        }
+    }
+
+    if (version.empty())
+    {
+        lg2::error("Unable to parse VERSION_ID out of /etc/os-release");
+        // let it hash the empty string
+    }
+
+    return std::format(
+        "{:X}", static_cast<uint32_t>(std::hash<std::string>{}(version)));
```

Comment:
Hmmm. "std::hash does not guarantee that the result will be the same across computers, builds, or even executions of the same build on the same computer. Your only guarantee is that during one execution, objects that are equal have the same hash."
Not sure we can rely on std::hash to provide a consistent hash across both BMC's.

Reply:
It has to be the same in both this app and the CFAM daemon.  So far it's been working fine but I do have a  story open for using SHA512 in both like software-updater uses for its hashes.

So if possible I'd like to leave this as it is for now and then change both the SHA512 before they're actually used for comparing as part of enabling redundancy.

Reply:
nevermind, will fix now.

Reply:
done.